app/routes.py

```diff
@@ -1,2 +1,218 @@
+            return {
+                "details": "Invalid data"
+            }, 400
+        new_task = Task(title = req_body["title"], description = req_body["description"], completed_at = req_body["completed_at"])
+        db.session.add(new_task)
+        db.session.commit()
+        response_body = {
+            "task": new_task.to_json()
+        }
+        return jsonify(response_body), 201
+
+    elif request.method == "GET":
+        order = request.args.get("sort")
+        if order == "asc":
+            new_order = Task.query.order_by(Task.title.asc())
+            response = []
+            for task in new_order:
+                response.append(task.to_json())
+            return jsonify(response), 200
+
+        elif order == "desc":
+            new_order = Task.query.order_by(Task.title.desc())
+            response = []
+            for task in new_order:
+                response.append(task.to_json())
+            return jsonify(response), 200
+
+        all_tasks = Task.query.all()
+        response_body = []
+        for any_task in all_tasks:
+            response_body.append(any_task.to_json())
+        return jsonify(response_body), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET", "DELETE", "PUT"], strict_slashes=False)
+def dealing_with_id(task_id):
+    a_task = Task.query.get(task_id)
+
+    if a_task is None:
+        return make_response("", 404)
+    
+    if request.method == "GET":
+        return {
+            "task": a_task.to_json()
+        }, 200
+    
+    elif request.method == "PUT":
+        info = request.get_json()
+
+        a_task.title = info["title"]
+        a_task.description = info["description"]
+        a_task.completed_at = info["completed_at"]
+
+        db.session.commit()
+        return {
+            "task": a_task.to_json()
+        }, 200
+    
+    elif request.method == "DELETE":
+        db.session.delete(a_task)
+        db.session.commit()
+        return {
+            "details": f"Task {a_task.task_id} \"{a_task.title}\" successfully deleted"
+        }, 200
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def make_task_complete(task_id):
+    id = Task.query.get(task_id)
+
+    if id is None:
+        return make_response("", 404)
+
+    id.completed_at = datetime.now()
+    db.session.commit()
+    response_body = {
+        "task": id.to_json()
+    }
+
+    path = "https://slack.com/api/chat.postMessage"
+    text_response = f"Someone just completed the task {id.title}"
+    query_params = {
+        "channel": "task-notifications",
+        "text": text_response
+    }
+    header = {
+        "Authorization": f"Bearer {os.environ.get('SLACK_TOKEN')}"
+    }
+    response = requests.post(path, params=query_params, headers=header)
+    
+
+    return jsonify(response_body), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def make_task_incomplete(task_id):
+    id = Task.query.get(task_id)
```

Comment:
💡 Refactor tip: I'd be careful about naming this variable 'id' what `Task.query.get(task_id)` does is return a single task containing the `task_id` you specified in `get()` from your Task table.